### PR TITLE
Fix tool setting lock properties for SetupCameraTool

### PR DESCRIPTION
**Pull request by bsteinbk**
- Fix SetupCameraTool so key-in fields are properly enabled/disabled based on their associated 'lock' property
- add PropertyDescriptionHelper.buildLockPropertyDescription
- update ToolSettings and DialogItem documentation
- update ToolWithSettings tool to better demonstrate how to set up ToolSettings properties.

### CODE REVIEW

**Review by pmconne — CHANGES_REQUESTED** [review]
You need to run extract-api.

**Review by pmconne — COMMENTED** [review]
I defer to @bbastings.

File: common/api/imodeljs-editor-frontend.api.md [hunk 1]
```diff
@@ -87,8 +88,6 @@ export class CreateArcTool extends CreateOrContinuePathTool {
     // (undocumented)
     protected getArcNormal(ev: BeButtonEvent): Vector3d;
     // (undocumented)
-    protected _getMethodDescription: () => PropertyDescription;
```
Comment (bbastings) [inline common/api/imodeljs-editor-frontend.api.md:90]: Need to revert/re-run extract api after reverting changes to SketchTool.ts? Shouldn't be any changes to editor package right?

**Review by bbastings — APPROVED** [review]

**Review by pmconne — APPROVED** [review]

File: common/api/imodeljs-editor-frontend.api.md (hunk 1 above)
Comment (bsteinbk) [reply common/api/imodeljs-editor-frontend.api.md]: > Need to revert/re-run extract api after reverting changes to SketchTool.ts? Shouldn't be any changes to editor package right?

@bbastings - I just fixed, I forgot to clean and rebuild after reverting SketchTool.ts

**Review by DanEastBentley — APPROVED** [review]

**Pull request merged**